src/helpers/worker-hours-info.model.ts

```diff
@@ -15,7 +21,28 @@ import { TranslationHelper } from "./translations.helper";
 import { VerboseDateHelper } from "./verbose-date.helper";
 import { ContractType } from "../common-models/worker-info.model";
 
-interface WorkerInfoForCalculateWorkerHoursInfo {
+export const DEFAULT_NORM_SUBTRACTION = WORK_HOURS_PER_DAY;
+
+//#region models
+export enum OvertimeType {
+  MaxiumNotOvertimeHoursExceed = "Maxium not overtime hours exceed",
+  RevisionDifference = "revision difference",
+}
+
+interface WorkHoursInfoCalulationOptions {
+  considerOvetimeTypes: OvertimeType[];
+}
+
+const DEFAULT_WORK_HOURS_INFO_CALCULATION_OPTIONS = {
+  considerOvetimeTypes: [],
```

Comment:
Wpadłem na pomysł, że skoro my będziemy chcieli móc wyłączać i włączać poszczególne typy nadgodzin,  to miało by sens zrobić tak, że obsługa nadgodzin będzie  robiona mniej więcej w taki sposób

```js
handlers = {
   [MaxNotOvertime]: calculateMaxNotOvertime,
   [RevisionDiff]: calculateRevisionDiff
}
....
function calculateOvertim(considerOvertimeTypes) {
  return considerOvertimeTypes.reduce((acc, type) => acc + handlers[type]()
}
```
```

Reply:
Ładnie to wygląda i można to jak najbardziej stosować, ale w tym przypadku to trochę przerost formy nad treścią, bo tylko 2 handlery i jak już ustawimy to po pandemii to nie będziemy więcej tego zmieniać.